packages/validator/src/services/attestation.ts

```diff
@@ -276,4 +291,95 @@ export class AttestationService {
       }
     }
   }
+
+  /**
+   * Performs additional attestation aggregation tasks required if validator is part of distributed cluster
+   *
+   * 1. Exchange partial for combined selection proofs
+   * 2. Determine validators that should aggregate attestations
+   * 3. Mutate duty objects to set selection proofs for aggregators
+   * 4. Resubscribe validators as aggregators on beacon committee subnets
+   *
+   * See https://docs.google.com/document/d/1q9jOTPcYQa-3L8luRvQJ-M0eegtba4Nmon3dpO79TMk/mobilebasic
+   */
+  private async runDistributedAggregationSelectionTasks(
+    duties: AttDutyAndProof[],
+    slot: number,
+    signal: AbortSignal
+  ): Promise<void> {
+    const partialSelections: routes.validator.BeaconCommitteeSelection[] = duties.map(
+      ({duty, partialSelectionProof}) => ({
+        validatorIndex: duty.validatorIndex,
+        slot,
+        selectionProof: partialSelectionProof as BLSSignature,
+      })
+    );
+
+    this.logger.debug("Submitting partial beacon committee selection proofs", {slot, count: partialSelections.length});
+
+    const res = await Promise.race([
+      this.api.validator
+        .submitBeaconCommitteeSelections(partialSelections)
+        .catch((e) => this.logger.error("Error on submitBeaconCommitteeSelections", {slot}, e)),
```

Comment:
Added this catch handler to log `submitBeaconCommitteeSelections` errors if thrown after cutoff slot time (1/3 of slot) as it seems like `Promise.race` would swallow that error if it happens after `sleep(this.clock.msToSlot(slot + 1 / 3), signal)` is already resolved.

Since we handle timeouts now via `ApiError.assert` (https://github.com/ChainSafe/lodestar/pull/5331) I am wondering if there can be an error that would even throw after 4 seconds (1/3 of slot).

What might happen now if there is an unexpected error (not even sure what error that would be though) and resolve before 1/3 of slot but without result which will throw `throw new Error("submitBeaconCommitteeSelections did not resolve after 1/3 of slot");` even though might not technically what happens.

I am bit split if this catch handler here is required, any thoughts @wemeetagain @matthewkeil?

Reply:
keeping additional catch handler to not swallow errors as outlined in comment above